### ISSUE TITLE
Investigate stability of quicksort_palette shader

### ISSUE THREAD

**Issue by afd** [google/graphicsfuzz#750]
My Intel GPU driver under Windows gives a different image for this shader compared with Swiftshader, and I have seen a fair bit of variation for this one over some fuzzing runs.

Interestingly the colors are fairly stable, but where the blocks of colors end up going varies.

This may just be due to shader compiler bugs, but it would be good to investigate by.

Image produced by Swiftshader:

![Image produced by Swiftshader](https://user-images.githubusercontent.com/4739677/65175186-4b767400-da4a-11e9-9700-ddc158e86da4.png)

Image produced by Intel Windows:

![Image produced by Intel Windows](https://user-images.githubusercontent.com/4739677/65175204-5204eb80-da4a-11e9-80b9-096b3e50e9bb.png)

**Comment by jarikomppa** [comment]
The puzzelize() causes a slight difference between float and double. I'll look closer..

**Comment by jarikomppa** [comment]
Well, this is a fun one.

Puzzelize calls randomize where the actual fun happens. This does the following operations:
```
dot
sin
mul
fract
add
floor
```
By tracking the result of each operation in both float and double, we can see where the problem happens (for a random failing case):
```
dot   ->  177.50000000  177.50000000
sin   ->    1.00000000    1.00000000
mul   -> 4250.00000000 4249.99999952
fract ->    0.00000000    0.99999952
add   ->    0.50000000    1.49999952
floor ->    0.00000000    1.00000000
```
By adding a magic constant after the mul, things get more stable. I reduced the float mantissa by masking bottom bits and got to 20 bit mantissa with the value 0.01; at 19 bits I couldn't find a single value that would fix all cases, but 0.02 had the least failing cases. 

So suggested change would be:
```
return float(floor(fract(sin(dot(co.xy ,vec2(12.5, 3.))) * 4250.) + 0.5));
->
return float(floor(fract(sin(dot(co.xy ,vec2(12.5, 3.))) * 4250. + 0.02) + 0.5));
```

Even after this we're slightly mathematically iffy, but having around 3 bits of error should be more than enough, even with the undefined rounding mode of the spec. Making the grid smaller produced more errors in the masked-mantissa modes, but not with the full float vs double when the constant was added.
